src/main/java/de/tum/in/www1/artemis/service/QuizSubmissionService.java

```diff
@@ -215,34 +221,20 @@ public QuizSubmission saveSubmissionForExamMode(QuizExercise quizExercise, QuizS
             throw new EntityNotFoundException("Participation for quiz exercise " + quizExercise.getId() + " and quiz submission " + quizSubmission.getId() + " for user "
                     + user.getLogin() + " was not found!");
         }
-        StudentParticipation studentParticipation = optionalParticipation.get();
-        quizSubmission.setParticipation(studentParticipation);
-        // remove result from submission (in the unlikely case it is passed here), so that students cannot inject a result
-        quizSubmission.setResults(new ArrayList<>());
-        quizSubmissionRepository.save(quizSubmission);
-
-        // versioning of submission
-        try {
-            submissionVersionService.saveVersionForIndividual(quizSubmission, user);
-        }
-        catch (Exception ex) {
-            log.error("Quiz submission version could not be saved", ex);
-        }
-
-        log.debug("submit exam quiz finished: {}", quizSubmission);
-        return quizSubmission;
+        return optionalParticipation.get();
     }
 
     /**
-     * Returns true if student has submitted at least once for the given quiz batch
+     * Save the quiz submission to the database and also save the submission version
      *
-     * @param quizBatch the quiz batch of interest to check if submission exists
-     * @param login     the student of interest to check if submission exists
-     * @return boolean the submission status of student for the given quiz batch
+     * @param quizExercise   the QuizExercise for which the given quizSubmission to be submitted
+     * @param quizSubmission the QuizSubmission to be saved
+     * @param user           the User by which the given quizSubmission was made
+     * @return saved QuizSubmission
      */
-    public boolean isSubmitted(QuizBatch quizBatch, String login) {
-        Set<QuizSubmission> submissions = quizSubmissionRepository.findAllByQuizBatchAndStudentLogin(quizBatch, login);
-        Optional<QuizSubmission> submission = submissions.stream().findFirst();
-        return submission.map(QuizSubmission::isSubmitted).orElse(false);
+    @Override
+    protected QuizSubmission save(QuizExercise quizExercise, QuizSubmission quizSubmission, User user) {
+        quizSubmission.setParticipation(this.getParticipation(quizExercise, quizSubmission, user));
```

Comment:
Maybe add a comment why the participation is set here? Calling this method it is not clear why you would have to provide the quizExercise and the user to save the quizSubmission.

Reply:
Yes, agree. The JavaDoc is updated.